validator/github.py

```diff
@@ -94,8 +103,15 @@ def get_custom_path(data):
     return data['content']['source']['path']
 
 
-def file_exists_on_repo(dockerfile, url, branch):
-    dockerfile_url = '{}/blob/{}/{}'.format(url, branch, dockerfile)
+# if token exist then using github API
+# https://developer.github.com/v3/repos/contents/#get-contents
+# GET /repos/:owner/:repo/contents/:path
+def file_exists_on_repo(dockerfile, url, branch, token):
+    if token is None:
+        dockerfile_url = '{}/blob/{}/{}'.format(url, branch, dockerfile)
+    else:
+        dockerfile_url = \
+            '{}/contents/{}/?ref={}'.format(url, dockerfile, branch)
```

Comment:
Just a note to self and others who might wonder... this would be the reason why we need the token. We could check for existence of branches with `git ls-remote` but I don't think there's another way to check for existence of dockerfile on that branch.